develop/parachain-devs/runtime-development/FRAME/add-pallet-to-the-runtime.md

```diff
@@ -0,0 +1,122 @@
+
+To ensure that the new dependencies resolve correctly for the runtime, you can run the following command:
+
+```bash
+cargo check -p solochain-template-runtime --release
+```
+
+## Config Trait for Pallets
+
+Every Polkadot SDK pallet defines a [Rust trait](https://doc.rust-lang.org/book/ch10-02-traits.html){target=\_blank} called `Config`. This trait specifies the types and parameters that the pallet needs to integrate with the runtime and perform its functions. The primary purpose of this trait is to act as an interface between this pallet and the runtime in which it is embedded. A type, function, or constant in this trait is essentially left to be configured by the runtime that includes this pallet.
+
+Consequently, a runtime that wants to include this pallet must implement this trait.
+
+You can inspect any pallet’s `Config` trait by reviewing its Rust documentation or source code. The `Config` trait ensures the pallet has access to the necessary types (like events, calls, or origins) and integrates smoothly with the rest of the runtime.
+
+At its core, the `Config` trait typically looks like this:
+
+```rust
+--8<-- 'code/develop/parachain-devs/runtime-development/FRAME/add-pallet-to-the-runtime/pallet-basic-config-trait.rs'
+```
+
+This basic structure shows that every pallet must define certain types, such as `RuntimeEvent` and `WeightInfo`, to be functional within the runtime. The actual implementation can vary depending on the pallet’s specific needs.
+
+??? "Example - Utility Pallet"
+      For instance, in the [`utility pallet`](https://github.com/paritytech/polkadot-sdk/tree/master/substrate/frame/utility){target=\_blank}, the Config trait is implemented with the following types:
+
+      ```rust
+      --8<-- 'code/develop/parachain-devs/runtime-development/FRAME/add-pallet-to-the-runtime/utility-pallet-config-trait.rs'
+      ```
+
+     This example shows how the `Config` trait defines types like `RuntimeEvent`, `RuntimeCall`, `PalletsOrigin`, and `WeightInfo`, which the pallet will use when interacting with the runtime.
+
+## Parameters Configuration for Pallets
+
+Traits in Rust define shared behavior, and within the Polkadot SDK, they allow runtimes to integrate and utilize a pallet's functionality by implementing its associated configuration trait and parameters. Some of these parameters may require constant values, which can be defined using the [`parameter_types!`](https://paritytech.github.io/polkadot-sdk/master/frame_support/macro.parameter_types.html){target=\_blank} macro. This macro simplifies development by expanding the constants into the appropriate struct types with functions that the runtime can use to access their types and values in a consistent manner.
+
+For example, the following code snippet shows how the Solochain template configures some parameters through the `parameter_types!` macro:
```

Comment:
Here we could link the line of the src code of the Solochain template where this is configured

Reply:
Same as above

Reply:
But here, you are mentioning a specific example, so we could link the repository with a specific tag as we already have this in the `variables.yaml` file

Reply:
yeah that makes sense. CC @eshaben just FYI